File: TREE/Java/LC 94 - BINARY TREE INORDER TRAVERSAL.java
// LEETCODE 94 - BINARY TREE INORDER TRAVERSAL
// QUESTION LINK - "https://leetcode.com/problems/binary-tree-inorder-traversal/"

import java.util.*;

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int x) { val = x; }
}

class Solution {
    private void inOrder(TreeNode root, List<Integer> res) {
        if (root == null) return;
        inOrder(root.left, res);
        res.add(root.val);
        inOrder(root.right, res);
    }

    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        inOrder(root, res);
        return res;
    }
}

File: TREE/Python/LC 94 - BINARY TREE INORDER TRAVERSAL.py
# LEETCODE 94 - BINARY TREE INORDER TRAVERSAL
# QUESTION LINK - "https://leetcode.com/problems/binary-tree-inorder-traversal/"

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def inOrder(self, root, res):
        if not root:
            return
        self.inOrder(root.left, res)
        res.append(root.val)
        self.inOrder(root.right, res)

    def inorderTraversal(self, root):
        res = []
        self.inOrder(root, res)
        return res
